### PR TITLE
added test case #2749

### LINKED ISSUE #2749: IndexError: list index out of range with as_explicit on an inverse
(from http://stackoverflow.com/q/20884626/161801)

```
In [20]: y=MatrixSymbol('y',5,1)
x=Ma
In [21]: x=MatrixSymbol('x',5,2)
b
In [22]: b=MatrixSymbol('b',2,1)

In [23]: (x.T*x).I.as_explicit()
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-23-93739c34be6e> in <module>()
----> 1 (x.T*x).I.as_explicit()

/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/matrices/expressions/matexpr.py in as_explicit(self)
    243         return ImmutableMatrix([[    self[i, j]
    244                             for j in range(self.cols)]
--> 245                             for i in range(self.rows)])
    246
    247     def as_mutable(self):

/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/matrices/expressions/matexpr.py in <listcomp>(.0)
    243         return ImmutableMatrix([[    self[i, j]
    244                             for j in range(self.cols)]
--> 245                             for i in range(self.rows)])
    246
    247     def as_mutable(self):

/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/matrices/expressions/matexpr.py in <listcomp>(.0)
    242         from sympy.matrices.immutable import ImmutableMatrix
    243         return ImmutableMatrix([[    self[i, j]
--> 244                             for j in range(self.cols)]
    245                             for i in range(self.rows)])
    246

/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/matrices/expressions/matexpr.py in __getitem__(self, key)
    211             i, j = sympify(i), sympify(j)
    212             if self.valid_index(i, j) is not False:
--> 213                 return self._entry(i, j)
    214             else:
    215                 raise IndexError("Invalid indices (%s, %s)" % (i, j))

/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/matrices/expressions/matpow.py in _entry(self, i, j)
     30         if self.exp.is_Integer:
     31             # Make an explicity MatMul out of the MatPow
---> 32             return MatMul(*[self.base for k in range(self.exp)])._entry(i, j)
     33
     34 from .matmul import MatMul

/Users/aaronmeurer/Documents/Python/sympy/sympy/sympy/matrices/expressions/matmul.py in _entry(self, i, j, expand)
     47             return coeff * matrices[0][i, j]
     48
---> 49         head, tail = matrices[0], matrices[1:]
     50         if len(tail) == 0:
     51             raise ValueError("lenth of tail cannot be 0")

IndexError: list index out of range
```

### PR DESCRIPTION
**Pull request by RituRajSingh878**
#### References to other Issues or PRs

Fixes #2749
#### Brief description of what is fixed or changed
Issue is solved in sympy master and works fine but not tested.
So i add a test case for this issue.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/matrices/expressions/tests/test_matexpr.py [hunk 1]
```diff
@@ -347,3 +347,8 @@ def test_factor_expand():
     assert expr1 != expr2
     assert expand(expr1) == expr2
     assert factor(expr2) == expr1
+
+def test_issue_2749():
+    A = MatrixSymbol("A", 5, 2)
+    assert (A.T*A).I.as_explicit()==Matrix([[(A.T*A).I[0, 0], (A.T*A).I[0, 1]], \
```
Comment (oscargus) [inline sympy/matrices/expressions/tests/test_matexpr.py]: Some space around == would be good. (And maybe around *, it looks quite crammed now, but most of the time there are no spaces around *.)

**Comment by oscargus** [comment]
I'll edit the release note. For these types of fixes it is better to just use `NO ENTRY` as they are directed to end users. For them, a test is not that interesting, especially when it points to "above issue" in this file containing probably hundreds of issues https://github.com/sympy/sympy/wiki/Release-Notes-for-1.4

Thanks! It will be merged in conjunction with something else to avoid extensive testing, but from my perspective it is good to go!

**Comment by smichr** [comment]
> It will be merged in conjunction with something

Oh...do you mean you will wait until their are two things to merge and then cancel the one build and let the other merge complete?

**Comment by oscargus** [comment]
Yes, something like that. I try to apply that for these types of minor things* to save a bit of CPU cycles... I was thinking about adding a category as well, so that people can merge these as well when they are merging other stuff. Not sure if it is worthwhile though, the current risk is that one forgets about it...

* Typically minor documentation fixes and additional tests.

**Pull request merged**